matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/session/permalinks/PermalinkParser.kt

```diff
@@ -65,17 +68,58 @@ object PermalinkParser {
                 PermalinkData.RoomLink(
                         roomIdOrAlias = identifier,
                         isRoomAlias = false,
-                        eventId = extraParameter.takeIf { !it.isNullOrEmpty() && MatrixPatterns.isEventId(it) }
+                        eventId = extraParameter.takeIf { !it.isNullOrEmpty() && MatrixPatterns.isEventId(it) },
+                        viaParameters = viaQueryParameters
                 )
             }
             MatrixPatterns.isRoomAlias(identifier) -> {
                 PermalinkData.RoomLink(
                         roomIdOrAlias = identifier,
                         isRoomAlias = true,
-                        eventId = extraParameter.takeIf { !it.isNullOrEmpty() && MatrixPatterns.isEventId(it) }
+                        eventId = extraParameter.takeIf { !it.isNullOrEmpty() && MatrixPatterns.isEventId(it) },
+                        viaParameters = viaQueryParameters
                 )
             }
             else                                   -> PermalinkData.FallbackLink(uri)
         }
     }
+
+    private fun String.getViaParameters(indexOfQuery: Int): List<String> {
+        val query = try {
+            substring(indexOfQuery + 1)
+        } catch (e: IndexOutOfBoundsException) {
+            return emptyList()
+        }
+        val encodedKey = try {
+            URLEncoder.encode("via", "UTF-8")
+        } catch (e: UnsupportedEncodingException) {
+            return emptyList()
+        }
+        val values = ArrayList<String>()
+        var start = 0
+        do {
+            val nextAmpersand = query.indexOf('&', start)
```

Comment:
This algo gives me headache :/
Maybe use split() method to improve it like you already did here: https://github.com/vector-im/element-android/blob/8c150aa3e8c97ae9794921b786810208ad92ced1/vector/src/main/java/im/vector/app/features/call/conference/JitsiWidgetProperties.kt#L30

Reply:
Same it's c/p from uri code... but can change